### DISCUSSION TITLE
Azure Repos: Authenticate Git with a PAT

### DISCUSSION THREAD

**Discussion by giscus** [rajbos/rajbos.github.io#30] [Comments]
# Azure Repos: Authenticate Git with a PAT

Azure Repos: Authenticate Git with a PAT

https://devopsjournal.io/blog/2020/11/08/Azure-DevOps-Git-Authenticate-With-PAT

**Comment by rl4518** [comment]
Hey, does this actually work?  Where is the function SetupAuthentication reside? Is this is in a git config file somewhere? (I'm not a git expert obviously, I'm just trying to get anything to work which will allow me to authenticate to azure devops.
the latest attempt:

 git clone https://rl4518:<PAT>@dev.azure.com/ORG/PROJECT/_git/REPO.repo fails in Visual Studio with "Recv failure: Connection was reset"

**Reply by rajbos** [reply]
Hi @rl4518 , `SetupAuthentication` is my own PowerShell function that I created since I needed to Base64 encode the token as well.
If you want to use this with Visual Studio, I cannot send those extra headers so you can skip this function and use one of the following methods (as mentioned in the post):

1. Solution (project based)
2. Solution (organization based)

**Comment by rl4518** [comment]
Thanks!

**Comment by psulek** [comment]
Thank YOU a million times! You save me from 401 not autorized errors on our company on-premise azure devops when connected to company via VPN !

**Reply by rajbos** [reply]
Great that this was helpful 🎉.
Thanks for letting me know!
